Assets/Game/Scripts/MapGeneration/Algorithms/BSP.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = System.Random;
+
+public class BSP : MapGenerator
```

Comment:
What is the goal using BSP here?

Reply:
I wrote it before we decided to use single path, same as Random Walk and Celluar Automata, I keep it there just in case we need it in the future.